### PR TITLE
Version update v0 12 1

### PR DESCRIPTION
**Pull request by graemedavidson**
- Updates module to be compatible with version 0.12.1 of Influxdb

- Adds basic rspec testing (matches puppet skeleton module)

- attempt to add acceptance tests via beaker, currently failing (see readme)

### CODE REVIEW

**Comment by graemedavidson** [comment]
@sedan07 - please review

File: lib/puppet/type/influxdb_user.rb [hunk 1]
```diff
@@ -50,29 +57,38 @@ def insync?(is)
...
+      is.to_s == should.to_s
+    end
   end
 
-  newparam(:influx_url)do
+  newparam(:influx_url) do
     desc "The URL of the influx HTTP API"
     defaultto 'http://localhost:8086'
+    validate do |value|
+      raise Puppet::ResourceError, "URL is invalid: '#{value}'" if value !~ /\A((http|https):\/\/)?([a-z\d])([a-z\d-]{0,61}\.)*[a-z\d]*(:\d{2,5})?$/
```
Comment (sedan07) [inline lib/puppet/type/influxdb_user.rb]: As hostnames are case insensitive this should support uppercase chars too

File: spec/acceptance/version.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+_osfamily               = fact('osfamily')
+_operatingsystem        = fact('operatingsystem')
+_operatingsystemrelease = fact('operatingsystemrelease').to_f
+
+case _osfamily
+when 'RedHat'
```
Comment (sedan07) [inline spec/acceptance/version.rb]: shouldn't the when be indented in?

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.project
+Gemfile.lock
+pkg
+coverage
+submodules
+.vagrant
+log
+junit
```
Comment (sedan07) [inline .gitignore]: Do we need this?

**Pull request merged**